smartsim/_core/entrypoints/dragon_client.py

```diff
@@ -0,0 +1,116 @@
+# BSD 2-Clause License
+#
+# Copyright (c) 2021-2024, Hewlett Packard Enterpris
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import argparse
+import json
+import os
+import signal
+import sys
+import time
+import typing as t
+
+import zmq
+
+from smartsim._core.launcher.dragon.dragonConnector import DragonConnector
+from smartsim._core.schemas import (
+    DragonHandshakeRequest,
+    DragonRequest,
+    DragonShutdownRequest,
+    request_registry,
+)
+from smartsim.log import get_logger
+
+SIGNALS = [signal.SIGINT, signal.SIGQUIT, signal.SIGTERM, signal.SIGABRT]
+
+logger = get_logger("Dragon Client")
+
+
+def cleanup() -> None:
+    logger.debug("Cleaning up")
+
+
+def main(args: argparse.Namespace) -> int:
+
+    requests: t.List[DragonRequest] = []
+
+    try:
+        with open(args.submit, "r", encoding="utf-8") as request_file:
```

Comment:
Can we turn this parsing of serialized requests into something that is testable outside the main function? It would be nice to know that we can replace a text-based "batch builder" or even just to reduce the amount of stuff main seems to be doing.